core/email_api.php

```diff
@@ -1395,3 +1395,51 @@ function email_build_visible_bug_data( $p_user_id, $p_bug_id, $p_message_id ) {
 
 	return $t_bug_data;
 }
+
+/**
+ * Registers shutdown function to send queued emails after script execution
+ * @return void
+ */
+function email_shutdown_function_register() {
+	$t_register = false;
+	if( php_sapi_name() == 'cli' ) {
+		# Skip processing for predefined scripts
+		switch( basename( $_SERVER['SCRIPT_NAME'] ) ) {
+			case 'send_emails.php':
```

Comment:
How about having the ability for any script page to SKIP_EMAIL_SENDING and having this code just check for that rather than specific script names?  We should also use the same technique to skip generation for scenarios where we generate secondary resources like css, javascript, etc.